ocs_ci/framework/pytest_customization/ocscilib.py

```diff
@@ -183,3 +192,17 @@ def pytest_collection_modifyitems(session, config, items):
                 f"{item.name} in {item.fspath}",
                 exc_info=True
             )
+
+
+@pytest.hookimpl(tryfirst=True, hookwrapper=True)
+def pytest_runtest_makereport(item, call):
```

Comment:
@vavuthu  how long does it take to get all data from must gather?

Vijay or @nehaberry do you know if we will get some advantage to run it after each failed test case? 

Isn't enough to run only at the end of the execution to get all the logs after the the execution and run it only once?  I guess that if if we will have 2 failed test cases that we will spend time to collecting logs twice with same results as we do only once at the end of execution or do I miss something?

**I see advantage if**: 

- must gather collect logs just from last execution of must gather of if we can specify somehow timestamp
- if we will miss some data from some deleted pods from the test which are not able to collect after some time . (This I am not sure how it works)

@nehaberry or @vavuthu do you know if must gather collect also logs for deleted pods?

Vijay, as far as I understand this hook is running after the test finish and it runs after finalizers of fixtures right?  Just thinking if must gather doesn't collect the logs from deleted pods that it probably will miss some logs from resources we will delete in finalizer. But not sure how it works.

Thanks for clarifications

Reply:
@petr-balogh ran must gather couple of times and  I can see it took ~31 sec
$ time oc adm must-gather --image=ashishranjan738/ocs-must-gather:latest

real	0m31.134s
user	0m0.723s
sys	0m0.259s

It will collect logs before tear down executes ( after test case fails but before finalizers executes ) so that we will have logs exactly in the state where test case failed which will be easy for debugging.

I am not sure whether it will collect deleted pods or not ( need to check )

Reply:
Great, if it will run after finalizers (tear down) that it's good addition and ~31 sec is not to much.  We will see if it will takes more after some longer running execution.

But that case I agree that it's better that all the logs will be more readable if they will end with the failure of the test case and won't have additional logs with teardown.

Thanks